examples/fluids/qfunctions/channel.h

```diff
@@ -29,46 +30,70 @@ struct ChannelContext_ {
   struct NewtonianIdealGasContext_ newtonian_ctx;
 };
 
-CEED_QFUNCTION_HELPER CeedInt Exact_Channel(CeedInt dim, CeedScalar time,
-    const CeedScalar X[], CeedInt Nf, CeedScalar q[], void *ctx) {
+CEED_QFUNCTION_HELPER State Exact_Channel(CeedInt dim, CeedScalar time,
+    const CeedScalar X[], CeedInt Nf, void *ctx) {
 
   const ChannelContext context = (ChannelContext)ctx;
-  const CeedScalar theta0 = context->theta0;
-  const CeedScalar P0     = context->P0;
-  const CeedScalar umax   = context->umax;
-  const CeedScalar center = context->center;
-  const CeedScalar H      = context->H;
-  const CeedScalar cv     = context->newtonian_ctx.cv;
-  const CeedScalar cp     = context->newtonian_ctx.cp;
-  const CeedScalar Rd     = cp - cv;
-  const CeedScalar mu     = context->newtonian_ctx.mu;
-  const CeedScalar k      = context->newtonian_ctx.k;
-
-  const CeedScalar y=X[1];
-
+  const CeedScalar theta0      = context->theta0;
+  const CeedScalar P0          = context->P0;
+  const CeedScalar umax        = context->umax;
+  const CeedScalar center      = context->center;
+  const CeedScalar H           = context->H;
+  NewtonianIdealGasContext gas = &context->newtonian_ctx;
+  const CeedScalar cp          = gas->cp;
+  const CeedScalar mu          = gas->mu;
+  const CeedScalar k           = gas->k;
+  // There is a gravity body force but it is excluded from
+  //   the potential energy due to periodicity.
+  gas->g[0] = 0.;
+  gas->g[1] = 0.;
+  gas->g[2] = 0.;
+
+  const CeedScalar y     = X[1];
   const CeedScalar Pr    = mu / (cp*k);
   const CeedScalar Ec    = (umax*umax) / (cp*theta0);
   const CeedScalar theta = theta0*(1 + (Pr*Ec/3)
                                    * (1 - Square(Square((y-center)/H))));
+  CeedScalar Y[5] = {0.};
+  Y[0] = P0;
+  Y[1] = umax*(1 - Square((y-center)/H));
+  Y[2] = 0.;
+  Y[3] = 0.;
+  Y[4] = theta;
+
+  return StateFromY(gas, Y, X);
+}
 
-  const CeedScalar p = P0;
+// *****************************************************************************
+// These QFunctions sets the initial condition
+// *****************************************************************************
+CEED_QFUNCTION(ICsChannel_Prim)(void *ctx, CeedInt Q,
```

Comment:
Since `ExactChannel` is outputing a `State`, I think it makes sense to add primitive to the newtonian context and have a single ICs Channel, and the output of it is either `s.Y.*` or `s.U.*`.